### DISCUSSION TITLE
Is there a way to pin the MDS port?

### DISCUSSION THREAD

**Discussion by caffeinism** [rook/rook#10790] [Q&A] [answered]
hello? I'm still a beginner in ceph.

A problem with my current server configuration gave me a reason to pin the MDS to a specific port. However, to my shallow knowledge, ports in MDS are not fixed, but are randomly assigned (but assigned sequentially) the same as ports in OSD.

But can you give it a different port band than the OSD's band? I also deploy MDS on the storage server, so I can't filter by IP.

**Comment by travisn** [comment]
Try the [Ceph Networking Reference](https://docs.ceph.com/en/latest/rados/configuration/network-config-ref/) for options available for Ceph daemons. See also a related Rook topic on [OSD network config](https://rook.io/docs/rook/latest/Storage-Configuration/Advanced/ceph-configuration/#osd-dedicated-network).

**Reply by caffeinism** [reply]
I've looked through those documents before, but unfortunately the port areas of OSD and MDS are integrated and cannot be separated. Is there still no way to separate it?

**Comment by caffeinism** [comment, marked as answer]
This was an easy problem to solve by separating the settings for OSD and MDS.

**Reply by travisn** [reply]
Where did you separate them? In the ceph.conf overrides?

**Reply by caffeinism** [reply]
> Where did you separate them? In the ceph.conf overrides?

yes

In the first time, I didn't realize I could seperate this config per each daemon.. I thought it can be configure global only.

Now, I wrote ceph.config like this
```
[osd]
ms_bind_port_...
[mds]
ms_bind_port_...
```
